velox/exec/tests/utils/AssertQueryBuilder.cpp

```diff
@@ -182,7 +183,11 @@ AssertQueryBuilder::readCursor() {
 
   if (!configs_.empty()) {
     if (!params_.queryCtx) {
-      params_.queryCtx = core::QueryCtx::createForTest();
+      if (executor_ == nullptr) {
+        executor_ = std::make_shared<folly::CPUThreadPoolExecutor>(
```

Comment:
This is not sufficient. Consider this code:

```
auto results = AssertQueryBuilder(plan).copyResults(pool());
```

Here, a temporary instance of AssertQueryBuilder is created, it makes new executors, starts a Task using that executor, copies the results, then deletes the executor's reference while Task is still running. This leads to the original problem where the Driver's destructor triggers Task's destructor, then QueryCtx's destructor, then Executor's destructor which blocks indefinitely.

The fix I made locally is the following:

```
  std::shared_ptr<folly::CPUThreadPoolExecutor> executor;
  auto guard = folly::makeGuard([&] {
    if (executor) {
      executor->join();
    }
  });

  if (!params_.queryCtx) {
    executor = std::make_shared<folly::CPUThreadPoolExecutor>(
        std::thread::hardware_concurrency());
    params_.queryCtx = std::make_shared<core::QueryCtx>(executor);
  }
  if (!configs_.empty()) {
    params_.queryCtx->setConfigOverridesUnsafe(std::move(configs_));
  }
```